File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Md5File.java
package com.onevgo.functions;

import cn.hutool.crypto.SecureUtil;

import java.io.File;

public class Md5File {
    public static String md5File(String filename) {
        return SecureUtil.md5(new File(filename));
    }

    public static void main(String[] args) {
        String file = "test.txt";
        System.out.println(md5File(file));
    }
}

File: 01_Language/01_Functions/go/src/functions/md5_file.go
package functions

import (
	"bufio"
	"crypto/md5"
	"encoding/hex"
	"os"
)

func Md5File(filename string) (string, error) {
	file, err := os.Open(filename)
	if err != nil {
		return "", err
	}
	defer file.Close()

	h := md5.New()
	reader := bufio.NewReader(file)
	b := make([]byte, 8192)
	for true {
		n, err := reader.Read(b)
		if err != nil {
			break
		}

		h.Write(b[0:n])
	}
	return hex.EncodeToString(h.Sum(nil)), nil
}
